boards/common/stm32f103c8/dist/openocd.cfg

```diff
@@ -1,4 +1,21 @@
+set WORKAREASIZE 0x5000
+
+set CHIPNAME STM32F103C8Tx
+
+# Enable debug when in low power modes
+set ENABLE_LOW_POWER 1
+
+# Stop Watchdog counters when halt
+set STOP_WATCHDOG 1
+
+# STlink Debug clock frequency
+set CLOCK_FREQ 4000
+
+# use hardware reset, connect under reset
+# connect_assert_srst needed if low power mode application running (WFI...)
+reset_config srst_only srst_nogate connect_assert_srst
```

Comment:
@MrKevinWeiss: I think that this change is kind of a regression when having the common Blue Pill and Black Pill user in mind: Most users likely just connect SWDIO, SWDCLK, GND and 3.3V to the device. In that case the user will now have to press the reset button manually to trigger flashing and once again after flashing to boot the new firmware.

Would it not make more sense to keep the default reset config, unless e.g. an environment variable like `STM32F103_USE_HW_RESET` is set to `1`?

Alternatively giving a hint in the documentation of the Blue Pill in the flash section that the debugger needs to be connected not only to SWDIO and SWDCLK, but also to the reset pin of the board would solve the issue to me. That way users won't start wondering why flashing using this OpenOCD config won't work without the debugger connected to the reset pin, while it does work fine with the default config of OpenOCD.

Reply:
Something that could be nice, would be to update the `openocd` handling so you could say `OPENOCD_FILE=openocd_hw_reset.cfg` or something. This way, no need for a board specific variable but it would be another task for the future.
Openocd has `--filename` for the config file and `--search` for the search path which would work nicely in that case.